src/trace/implementations/spine_fueled.rs

```diff
@@ -482,6 +482,12 @@ where
             }
         }
 
+        // If the logical frontier is empty, we are not obliged to be able to represent any time,
+        // since no time is in advance of `[]`. In the case, we can drop all contents.
+        if self.logical_frontier.is_empty() {
+            self.drop_batches();
+        }
+
```

Comment:
Is there an explainer for why the code is here, rather than in e.g. `set_logical_compaction`?

Reply:
Moved the code to `set_logical_compaction` and `insert`. As far as I can tell these are the only places where logical compaction is advanced, and new batches are inserted.